src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScorer.java

```diff
@@ -221,16 +225,27 @@ public static void writeMissingReferenceAccessions(final String path, final SAMF
         }
 
         //Find and return all alignments meeting the coverage/identity criteria
-        final Collection<String> hits = new ArrayList<>();
-        if (isValidAlignment(read.getCigar(), read.getAttributeAsInteger("NM"), minCoverage, minIdentity)) {
-            hits.add(SAMSequenceRecord.truncateSequenceName(read.getAssignedContig()));
+        final ArrayList<PSPathogenHitAlignment> hits = new ArrayList<>();
+        final double minIdentityBases = minIdentity * read.getLength();
+        final int numMismatches = read.getAttributeAsInteger("NM");
+        final int numMatches = HostAlignmentReadFilter.getMatchesLessDeletions(read.getCigar(), numMismatches);
+        if (numMatches >= minIdentityBases) {
+            final String recordName = SAMSequenceRecord.truncateSequenceName(read.getAssignedContig());
+            hits.add(new PSPathogenHitAlignment(numMatches, recordName, read.getCigar()));
         }
-        hits.addAll(getValidAlternateHits(read, "XA", 0, 2, 3, minCoverage, minIdentity));
-        hits.addAll(getValidAlternateHits(read, "SA", 0, 3, 5, minCoverage, minIdentity));
+        hits.addAll(getValidAlternateHits(read, "XA", 0, 2, 3, minIdentityBases));
```

Comment:
Is there SAMTag.SA and SAMTag.XA? I think there might be missing... not sure.

Reply:
Doesn't look like it has these